lib/engine/step/g_18_ga/buy_company.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require_relative '../buy_company'
+
+module Engine
+  module Step
+    module G18GA
+      class BuyCompany < BuyCompany
+        def process_buy_company(action)
+          super
+
+          return unless action.company.sym == 'OSR'
+
+          owner = action.company.owner
+          return if owner.player? || owner.trains.size == @game.phase.train_limit || @game.phase.available?('4')
+
+          free_two_train = @game.trains.select { |t| t.name == '2' }.last.dup
```

Comment:
this should be trains.find  { |t| t.name == '2' }.dup, but what happens if this train is owned by someone else already?

Reply:
why don't you just make a new train?

Reply:
Ok, making a new train